book/getting-started/run-a-wallaroo-application-docker.md

```diff
@@ -0,0 +1,201 @@
+
+NOTE: You might need to run Docker commands with sudo depending on how you set up Docker.
+
+Let's get started!
+
+## Terminal 1, Start the Wallaroo Docker container
+
+```bash
+docker run --rm -it --privileged -p 4000:4000 -v /tmp/wallaroo-docker/wallaroo-src:/src/wallaroo -v /tmp/wallaroo-docker/python-virtualenv:/src/python-virtualenv --name wally wallaroo-labs-docker-wallaroolabs.bintray.io/dev/wallaroo:7b1f3550
+```
+
+### Breaking down the Docker command
+
+- `docker run`: The Docker command that will start a new container.
+
+- `--rm`: This flag will automatically clean up the container and remove the file system on exit.
+
+- `-it`: This flag will allow us to work with interactive processes by allocating a tty for the container.
+
+- `--privileged`: This flag gives the container access to the hosts' devices. This allows us to take advantage of certain system calls used by Wallaroo, specifically `mbind` and `set_mempolicy`. This flag is optional, but by excluding it there will be a performance degradation in Wallaroo's processing capabilities.
+
+- `-p 4000:4000`: This flag will map the default port for HTTP requests for the Metrics UI from the container to the host. This will make it possible to call up the Metrics UI from a browser on the host.
+
+- `-v /tmp/wallaroo-docker/wallaroo-src:/src/wallaroo`: This flag will mount a host directory as a data volume within the container. The first time you run this, you must use an empty directory in order for the Docker image to copy the Wallaroo source code to your host. This will allow you to open and modify the Wallaroo source code with the editor of your choice on your host. The Wallaroo source code will persist on your machine after the container is stopped or deleted. This flag is optional, but without it you'd need to use an editor within the container to view or modify the Wallaroo source code.
+
+- `-v /tmp/wallaroo-docker/python-virtualenv:/src/python-virtualenv`: This flag will mount a host directory as a data volume within the container. The first time this is run for the provided directory, this command will setup a persistent Python Virtual Environment for the container on your host. Thus, if you need to install any python modules using pip or easy_install they will persist after the container is stopped or deleted. This flag is optional, but without it, you will not have a persistent Python Virtual Environment for the container.
+
+- `--name wally`: The name for the container. This flag is optional but makes it easier to reference the container when we open other Bash sessions in the container.
+
+## Terminal 2, Start the Metrics UI
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+This command will start a new Bash shell within the container, we enter via the `environment-setup.sh` script to ensure we're in our persistent Python Virtual Environment.
```

Comment:
"... within the container, which will run the `environment-setup.sh` script to ensure our persistent Python virtualenv is setup."